src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    public static final int INITIAL_SIZE = 10;
+
+    private T[] array = (T[]) new Object[INITIAL_SIZE];
+    private int size = 0;
+
     @Override
     public void add(T value) {
-
+        checking();
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        generateIndexException(index < 0);
+        checking();
+        if (index == size) {
+            add(value);
+        } else {
+            T t = array[index];
+            generateIndexException(t == null);
```

Comment:
Checking if `t == null` is not a proper way to validate the index. An index within the size range but not pointing to an actual element (e.g., when the list is not fully populated) is still a valid index.

Reply:
Don't understand